### PR TITLE
Add new security hotspots for TS

### LINKED ISSUE #1574: Add new security hotspots for TS
RSPEC-2255
RSPEC-5122
RSPEC-4787
RSPEC-4790
RSPEC-4721
RSPEC-4823
RSPEC-2245
RSPEC-4784
RSPEC-4818
RSPEC-2077
RSPEC-4829
RSPEC-4817

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
Fixes #1574

### CODE REVIEW

**Comment by quentin-jaquier-sonarsource** [comment]
The three last rules (S4787, S4817 and S2077), reported no issues on the ruling: 3 new files have been added to cover them.

Pay special attention to the last commit (S2077), I changed the profile in the IT's to target only the rules that we want to test, preventing others to interfere.

File: its/plugin/tests/src/test/resources/ts-rules-project-profile.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<profile>
+  <name>ts-rules- guioproject-profile</name>
```
Comment (yassin-kammoun-sonarsource) [inline its/plugin/tests/src/test/resources/ts-rules-project-profile.xml]: Did you want to write `ts-rules-project-profile` instead ?

**Comment by quentin-jaquier-sonarsource** [comment]
There is a job supposed to update the list, not sure exactly when it will run, but I will keep an eye on it.

**Comment by saberduck** [comment]
well done!

**Pull request merged**